integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/apiDeploy/TrailingSlashTestCase.java

```diff
@@ -0,0 +1,93 @@
+        headers.put(HttpHeaders.AUTHORIZATION, "Bearer " + jwtTokenProd);
+    }
+
+    @Test(description = "test path with trailing slash but without path parameters")
+    public void testTrailingSlashWithoutPathParam() throws CCTestException, MalformedURLException {
+        String endpoint = Utils.getServiceURLHttps(API_CONTEXT + "/pet/findByStatus");
+        HttpResponse response = HttpsClientRequest.doGet(endpoint, headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(response.getData(), ResponseConstants.RESPONSE_BODY, "Response Body Mismatch.");
+
+        String endpoint_with_slash = Utils.getServiceURLHttps(API_CONTEXT + "/pet/findByStatus/");
+        HttpResponse response_with_slash = HttpsClientRequest.doGet(endpoint_with_slash, headers);
+        Assert.assertNotNull(response_with_slash);
+        Assert.assertEquals(response_with_slash.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(response_with_slash.getData(), ResponseConstants.RESPONSE_BODY, "Response Body Mismatch.");
+    }
+
+    @Test(description = "test path with trailing slash with path parameter")
+    public void testTrailingSlashWithPathParam() throws CCTestException, MalformedURLException {
+        String endpoint = Utils.getServiceURLHttps(API_CONTEXT + "/pet/1");
+        HttpResponse response = HttpsClientRequest.doGet(endpoint, headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(response.getData(), ResponseConstants.PET_BY_ID_RESPONSE, "Response Body Mismatch.");
+
+        String endpoint_with_slash = Utils.getServiceURLHttps(API_CONTEXT + "/pet/1/");
+        HttpResponse response_with_slash = HttpsClientRequest.doGet(endpoint_with_slash, headers);
+        Assert.assertNotNull(response_with_slash);
+        Assert.assertEquals(response_with_slash.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(response_with_slash.getData(), ResponseConstants.PET_BY_ID_RESPONSE, "Response Body Mismatch.");
+    }
+
+    @Test(description = "test path with trailing slash with multiple path parameters")
+    public void testTrailingSlashWithMultiplePathParams() throws CCTestException, MalformedURLException {
+        String endpoint = Utils.getServiceURLHttps(API_CONTEXT + "/store/1/pet/123");
+        HttpResponse response = HttpsClientRequest.doGet(endpoint, headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(response.getData(), ResponseConstants.STORE_INVENTORY_RESPONSE, "Response Body Mismatch.");
+
+        String endpoint_with_slash = Utils.getServiceURLHttps(API_CONTEXT + "/store/1/pet/123");
```

Comment:
Here, it should end with a / ?

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/2877/commits/b5b41e57906ff728fcda6d305ef7195b7e992146